src/Parsers/ASTInterpolateElement.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <Parsers/IAST.h>
+
+
+namespace DB
+{
+
+class ASTInterpolateElement : public IAST
+{
+public:
+    String column;
+    ASTPtr expr;
+
+    String getID(char) const override { return "InterpolateElement"; }
+
+    ASTPtr clone() const override
+    {
+        auto clone = std::make_shared<ASTInterpolateElement>(*this);
+        clone->cloneChildren();
```

Comment:
hm, I think you need to fix ptr for `clone->expr`; otherwise it still points to old ast.

Reply:
Maybe like here
https://github.com/ClickHouse/ClickHouse/blob/5a1392a8e349c2bba7e35b01b3e4f2b3c095aeae/src/Parsers/ASTColumnDeclaration.cpp#L10-L48

Reply:
did something, but probably still wrong - will revisit tomorrow...